components/gitpod-db/go/oidc_client_config.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package db
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/google/uuid"
+	"gorm.io/datatypes"
+	"gorm.io/gorm"
+	"time"
+)
+
+type OIDCClientConfig struct {
+	ID uuid.UUID `gorm:"primary_key;column:id;type:char;size:36;" json:"id"`
+
+	Issuer string `gorm:"column:issuer;type:char;size:255;" json:"issuer"`
+
+	Data datatypes.JSON `gorm:"column:data;type:text;size:65535" json:"data"`
```

Comment:
Is the idea to store common properties such as `ClientID`, `ClientSecret, `RedirectURL` in the generic JSON structure? Also wouldn't we need to look up entries by domain?

Reply:
Issuer is the domain, as per the oidc spec

Reply:
Yes, for now we're storing them as json